Game.cpp

```diff
@@ -156,10 +156,24 @@ void Game::attackMonsters(int &remainingMonsters)
 
 void Game::print()
 {
-    for (int y = 0; y < map.getHeight(); y++) 
+    const int startY = heroY - hero->getLightRadius() > 0 ? heroY - hero->getLightRadius() : 0;
+    const int endY = heroY + hero->getLightRadius() + 1 < map.getHeight() ? heroY + hero->getLightRadius() + 1 : map.getHeight();
+
+    for (int y = startY; y < endY; y++) 
     {  
-        for (int x = 0; x < map.getWidth(y); x++) 
-        {    
+        const int startX = heroX - hero->getLightRadius() > 0 ? heroX - hero->getLightRadius() : 0;
+        const int endX = heroX + hero->getLightRadius() + 1 < map.getWidth(y) ? heroX + hero->getLightRadius() + 1 : map.getWidth(y);   
+
+        if(y == startY)
+        {
+            std::cout << "╔";
+            for(int i = 0; i < 2*hero->getLightRadius()+1; i++) std::cout << "══";
```

Comment:
A limitet ellenőrizni kellene, mert 2-nél működik, de amint nagyobb értékkel rendelkezik a `LightRadius` elcsúszik a keret. (Alsó keret is, egyébként jó)